### PR TITLE
Updates to readme

### PR DESCRIPTION
**Pull request by smarterclayton**
Further refine goals and intro

@bparees

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,59 +7,45 @@
...
 
-### Speed   
-With S2I, the assemble process can perform a large number of complex operations without creating a new layer at each step, resulting in a fast process. In addition, S2I scripts can be written to re-use artifacts stored in a previous version of the application image, rather than having to download or build them each time the build is run.
+### Reproducibility
+Allow build environments to be tightly versioned by encapsulating them within a Docker image and defining a simple interface (injected source code) for callers.
 
-### Patchability  
-S2I allows you to rebuild the application in a consistent way if an underlying image needs a patch due to a security issue.
+### Flexibility   
+The scripts that process the application source code can be injected into the builder image, allowing an author to adapt existing images.
```
Comment (bparees) [inline README.md]: Mention patchability here.

File: README.md [hunk 2]
```diff
@@ -7,59 +7,45 @@
 [![Travis](https://travis-ci.org/openshift/source-to-image.svg?branch=master)](https://travis-ci.org/openshift/source-to-image)
 [![License](https://img.shields.io/github/license/openshift/source-to-image.svg)](https://www.apache.org/licenses/LICENSE-2.0.html)
 
-Source-to-Image (`S2I`) is a tool for building reproducible Docker images. `S2I` produces
-ready-to-run images by injecting source code into a Docker image and *assembling*
-a new Docker image for the application.  The result is then ready to use with `docker run`. 
+Source-to-Image (S2I) is a toolkit and workflow for building reproducible Docker images from source code. S2I produces
+ready-to-run images by injecting source code into a Docker container and letting the container prepare that source code for execution. By creating these self-assembling **builder images**, you can version and control your build environments exactly like you use Docker images to version your runtime environments.
```
Comment (bparees) [inline README.md]: Confusing what "these builder images" refers to. It follows from the previous sentence which is talking about producing application images.

File: README.md [hunk 3]
```diff
@@ -7,59 +7,45 @@
...
+For example, to create a reproducible build pipeline for Tomcat (the popular Java webserver) and Maven:
 
-Try [building your own application image](#getting-started).
+1. Create a builder image containing OpenJDK and Tomcat that expects to have a WAR file injected
+2. Create a second image that layers on top of the first image Maven and any other standard dependencies, and expects to have a Maven project injected
+3. Invoke source-to-image using the Java application source and the Maven image to create the desired application WAR
+4. Invoke source-to-image a second time using the WAR file from the previous step and the Tomcat image to create the runtime image
```
Comment (bparees) [inline README.md]: "And the tomcat image from the first step."

File: README.md [hunk 4]
```diff
@@ -7,59 +7,45 @@
...
 
-### Operational security  
-Building an arbitrary Dockerfile exposes the host system to root privilege escalation. This can be exploited by a malicious user because the entire Docker build process is run as a user with Docker privileges. S2I restricts the operations that can be performed as a root user and can run user
-supplied build scripts as a non-root user.  Furthermore, operations teams can supply whitelisted builder images which will not perform
-arbitrary actions such as installing random packages or downloading content from untrusted sources.
+### Speed   
+Instead of building multiple layers in a single Dockerfile, S2I encourages (and benefits from) authors to set up an application in a single image layer. This is both more powerful and faster than classic multi-layer builds.
 
-### Ecosystem
-S2I encourages a shared ecosystem of images where you can leverage best practices for your applications and build processes.
```
Comment (bparees) [inline README.md]: We need to say something about s2i working with pretty much any existing image and you  can either add scripts to the image, or provide the scripts externally and use the image as is.

Comment (smarterclayton) [inline README.md]: That's in flexibility.

Comment (bparees) [inline README.md]: So it is.

**Comment by bparees** [comment]
squash+lgtm.

**Pull request merged**